Frameworks/UIKit.Xaml/ProgressRing.xaml.cpp

```diff
@@ -18,22 +18,121 @@
 #include "pch.h"
 #include "ProgressRing.xaml.h"
 
+#include "ObjCXamlControls.h"
+
+using namespace Microsoft::WRL;
+using namespace Platform;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+
 namespace UIKit {
-namespace Xaml {
+	namespace Xaml {
 
-ProgressRing::ProgressRing() {
-    InitializeComponent();
-}
+		DependencyProperty^ ProgressRing::s_isActiveProperty = nullptr;
+		DependencyProperty^ ProgressRing::s_foregroundProperty = nullptr;
+		DependencyProperty^ ProgressRing::s_heightProperty = nullptr;
+		DependencyProperty^ ProgressRing::s_widthProperty = nullptr;
+		Platform::Boolean ProgressRing::s_dependencyPropertiesRegistered = false;
+
+		ProgressRing::ProgressRing() {
+			InitializeComponent();
+			_RegisterDependencyProperties();
+		}
+
+		void ProgressRing::_RegisterDependencyProperties() {
+			if (!s_dependencyPropertiesRegistered) {
+				s_dependencyPropertiesRegistered = true;
+
+				/* NOTE: Currently we must attach DependencyProperties to FrameworkElement for testing
+						because type information for custom controls is not currently being exposed to
+						the testing framework.
+						See Issue #2607 for current status.
+				*/
+				// TODO: This DependencyProperty should be attached to UIKit::Xaml::ProgressRing
+				s_isActiveProperty = DependencyProperty::RegisterAttached("ProgressRing_IsActive",
+					Platform::Boolean::typeid,
+					FrameworkElement::typeid,
+					nullptr);
+				// TODO: This DependencyProperty should be attached to UIKit::Xaml::ProgressRing
```

Comment:
>// TODO: This DependencyProperty should be attached to UIKit::Xaml::ProgressRing [](start = 4, length = 80)

nit: keep one comment line with slightly rewording for clarity?  also, can use one newline as separator before the comment for clarity.

[https://github.com/Microsoft/WinObjC/wiki/Coding-Standards](https://github.com/Microsoft/WinObjC/wiki/Coding-Standards) is the general code conventions for islandwood.